playwright/pages/components/EnvironmentSelector.ts

```diff
@@ -0,0 +1,61 @@
+import { expect, Locator, Page } from "@playwright/test";
+
+export class EnvironmentSelector {
+  readonly page: Page;
+  readonly loginTextButton: Locator;
+  readonly loginIconButton: Locator;
+  readonly environmentName: Locator;
+  readonly dropdownTrigger: Locator;
+
+  constructor(page: Page) {
+    /**
+     * Components
+     */
+    this.page = page;
+
+    /**
+     * Static locators
+     */
+    this.loginTextButton = page.getByRole("button", {
+      name: "Login required",
+      exact: true,
+    });
+    // TODO: Replace with `getByRole` once `<IconButton>` supports labels.
+    this.loginIconButton = page.getByTestId("environment-login-icon-button");
+    this.environmentName = page.getByTestId("active-environment-name");
+    // TODO: Replace with `getByRole` once `<IconButton>` supports labels.
+    this.dropdownTrigger = page.getByTestId("environment-dropdown-button");
+  }
+
+  /**
+   * Dynamic locators
+   */
+  getEnvironmentMenuItem(name: string): Locator {
+    return this.page.getByRole("menuitem", { name, exact: true });
+  }
+
+  /**
+   * Actions
+   */
+  async selectEnvironment(name: string) {
+    await this.dropdownTrigger.click();
+    await this.getEnvironmentMenuItem(name).click();
+    await expect(this.environmentName).toHaveText(name);
+  }
+
+  /**
+   * Assertions
+   */
+  // TODO: Type `kind` with `@slicemachine/manager`'s `Environment["kind"]`.
+  async checkSelectedEnvironmentKind(kind: "prod" | "stage" | "dev") {
+    const humanReadableKindMap = {
+      prod: "Production",
+      stage: "Staging",
+      dev: "Development",
+    } as const;
+
+    await expect(this.page.getByRole("status")).toHaveText(
```

Comment:
⚠️ Dangerous since tomorrow with two "status" it would break, right? We could use something before instead of `page` to target the right element. 

Also, good one here to use `toHaveText` since it's not visible

Reply:
I replaced the status check with a more direct `data-cy` locator: https://github.com/prismicio/slice-machine/pull/1237#discussion_r1430798695